src/extension/rpc.ts

```diff
@@ -17,46 +17,34 @@ type RPCResponseMessageFormat =
   | { sourceId: number; result: unknown }
   | { sourceId: number; error: unknown };
 
-export type RPCMessage<
-  Name extends string,
-  Params extends RPCParams,
-  ReturnType,
-> = {
-  __name: Name;
-  __params: Params;
-  __returnType: ReturnType;
-};
+type MessageCollection = Record<string, (...parameters: [SafeAny]) => SafeAny>;
 
-export interface RpcClient<
-  Messages extends RPCMessage<string, RPCParams, SafeAny>,
-> {
-  request: <TName extends Messages["__name"]>(
+export interface RpcClient<Messages extends MessageCollection> {
+  request: <TName extends keyof Messages & string>(
```

Comment:
I'm curious the reason behind the `& string` here. Is that just to filter out any potential keys that aren't strings?

Reply:
Yeah, otherwise `keyof` will always be `| symbol`, which makes using this in other places very annoying.